### ISSUE TITLE
configuration issue

### ISSUE THREAD

**Issue by mthlvt** [woder/meross_offline#1]
Hi @woder, I'm looking into your meross_offline component which seems promising!

I've setup as follows:

- placed component code under /config/custom_components/meross_offline
- built config.json as instructed with 1 device that is reporting to MQTT
- placed config.json and ca.crt under /config/meross

I'm reloading HA but there's no trace of the component in the logs and nothing happens, is there a way to get more verbose logging?

Could you please detail a bit more configuration steps? I've read the blog post and readme here but I have some doubts about the setup.

Thanks!

**Comment by woder** [comment]
Hi,

This is certainly interesting. Normally, if there is no trace of the component in the log then that means HA isn't finding it. As an example, even if the integration has an error while starting up you should get something like this in your log.
`WARNING (MainThread) [homeassistant.loader] You are using a custom integration for meross_devices which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.`

This leads me to believe there is an error in where you put the component somehow. In my case, the path is more like "config\homeassistant\custom_components\meross_devices" (sorry, using the Samba share to get this). 

Basically, you need to be sure it's under custom_components, which is directly under the "homeassistant" folder. You can then put all the config files at the root of the "homeassistant" folder. It's difficult to explain because my instance has been running since Home Assistant's infancy, which means my folder structure could be different to yours. If you can confirm your putting in the right place (maybe by installing a different custom component?) then please let me know and we can debug further.

**Comment by mthlvt** [comment]
Thanks for the message.

Strange indeed, I've installed several custom_components already and I confirm the folder is in the right place:

custom_components\sonoff is found

> WARNING (MainThread) [homeassistant.loader] You are using a custom integration sonoff which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

custom_components\meross_offline is not found (no log)

I checked and it's not a file permission issues, both folder have same user/group ownership and readable by HA

Could you please specify which version of HA you are using ? mine is latest (2021.3.4)

**Comment by woder** [comment]
Hey,

Looks like I'm using version 2021.1.5, which isn't the latest but is still rather recent. I think it might be a good idea to enable logging for the component just to see if it's running into any errors. You can do that as follows:
`logger:
  default: info
  logs:
    meross_protocol: info
    Meross_Light: info`

That should work well. Now I was checking the code to see what could cause such an issue, and I noticed that a configuration problem could do it. Specifically, be sure that the 
`"ca_cert": "/config/custom_components/meross_offine/ca.crt"`
setting is correct. In my case, "/config" is actually a folder within the Python virtual env. I don't really think that's required at all, but that's how mine is configured (it's been a really long time, I don't completely remember early revisions to be honest). I also realized now that you need to give a path that will resolve within docker (if you are using supervised install).

Basically, my Home Assistant Core is running in a docker container and has the config folder mounted at /config/<here>. The difference between this component and one like Sonoff is that mine tries to resolve a file from the path given in the "meross_config.json". 

To summarize, add the log statements to catch any problems that we haven't thought of, then check your path to "ca_cert" within meross_config.json. You can then just check that you can access the **absolute path** from the shell. If doing something like "ls /config/custom_components/meross_offline/ca.crt" returns an error or has the wrong permissions, that's probably the problem. Be sure that your certificate uses an absolute path, I don't think it will work with a relative path.

Let me know how that goes.

**Comment by mthlvt** [comment]
could you please confirm the full path of the expected meross_config.json  is  /config/meross_config.json ?

**Comment by woder** [comment]
Yes, your path looks correct. The path should be in the root of Home Assistant Core, along with stuff like the folder "custom_components". Aka you should be able to see that directory from the root folder, which is where meross_config.json is meant to be.

**Comment by mthlvt** [comment]
Thanks for your feedback, few leads to explore:

- could it be that the code needs several config files in different places? 

for example in the code there are multiple references to config json files:
|file path | code related to config file|
|---------| -----------------------------|
| \config\custom_components\meross\config.py  | `CONFIG = "~/.config/meross/config.json"`|
| \config\custom_components\meross\climate.py  | `cfg = config.load(hass.config.path("meross_config.json"))`|
| \config\custom_components\meross\light.py  | `cfg = config.load(hass.config.path("meross_config.json"))`|
| \config\custom_components\meross\switch.py  | `cfg = config.load(hass.config.path("meross_config.json"))`|

- could it be that the component is not recognized due to a missing dependency? like merossIot for example

**Comment by woder** [comment]
It is certainly possible that everything stops working if a dependency is missing. MerossIoT is necessary for it to work, there's, unfortunately, no way around that so you'll want to make sure it's installed. Also, the code is not ideal and I think that the "CONFIG" variable is no longer used. 

That said, I'm pretty sure the code is looking for the config file in the hass.config.path, so unless yours is different this would work just fine. I would check the library next. It's such a pain because I could never get git requirements working. The plugin won't do anything unless the library is found, so be sure it's there.

Mine is a terrible hack, but it involves entering a console in the python virtual environment HA Core uses, then doing a pip install with the right version of MerossIoT. If the problem was the libs, this would fix it.
